azavea/raster-foundry
# Add nodes for input projects to diagram

## Overview

This PR adds nodes for each input within the lab diagram. This allows for the comparison of the original projects to their processed counterparts.

Also in this PR is a fix to the publish URL to only show port designations when necessary.

### Demo

![screen shot 2017-01-26 at 10 59 36 am](https://cloud.githubusercontent.com/assets/2442245/22339173/18d46a0c-e3b7-11e6-8b5c-0b45e69ab7e9.png)

Shown here is comparing a color-corrected project (using the forest-fire preset) and its NDVI.
![screen shot 2017-01-26 at 11 00 10 am](https://cloud.githubusercontent.com/assets/2442245/22339172/18d1d666-e3b7-11e6-8ffa-7f25130be72e.png)


## Testing Instructions

 * Check that the input nodes show up
 * Check that you can compare the original projects to each other or to any other node.
 * Run migrations and ensure that they run successfully

Connects #972, #990

EDIT: wrapped a migration for the NDVI tool and another small linting fix into this PR (see second commit)
